opal/corelib/file.rb

```diff
@@ -30,6 +30,18 @@ def basename(path)
       split(path)[-1]
     end
 
+    def extname(path)
+      raise TypeError, 'no implicit conversion of nil into String' if path.nil?
+      path = path.to_path if path.respond_to?(:to_path)
+      raise TypeError, "no implicit conversion of #{path.class} into String" unless path.is_a?(String)
+      filename = basename(path)
+      return '' if filename.nil?
+      last_dot_idx = filename[1..-1].rindex('.')
+      extname = last_dot_idx.nil? ? '' : filename[(last_dot_idx + 1)..-1]
+      return '' if extname.tr('.', '').empty? # extname contains only dots
```

Comment:
Yes I did. 
I've took a different approach, now I'm just checking the last dot index with the filename length.